lib/nn-syntax/tonn_converter.js

```diff
@@ -282,12 +282,37 @@ module.exports = class ToNNConverter {
             else
                 return this.findEntity('LOCATION', value);
         } else if (value.isDate) {
-            if (value.value === null)
+            if (value.value === null) {
                 return 'now';
-            else if (value.value instanceof Ast.DateEdge)
+            } else if (value.value instanceof Ast.DateEdge) {
                 return List.concat(value.value.edge, 'unit:' + value.value.unit);
-            else
+            } else if (value.value instanceof Ast.DatePiece) {
+                let toConcat = ['new', 'DatePiece', '('];
+                if (value.value.originalYear !== null) {
+                    if (isSmallInteger(value.value.originalYear))
+                        toConcat.push(String(value.value.originalYear));
```

Comment:
I wonder if we can do something smarter in this code and avoid having to track "originalYear". For example, for all years between 1950 and 2050 we can first check if the sentence has the four digit year (`findEntity(... , { ignoreNotFound: true })`, and then try look for the two digit year or represent as a small number.
Primarily because it would be weird that paraphrasing 2008 to '08 suddenly changed the ThingTalk representation in a way that we cannot automatically convert one to the other.